### PR TITLE
Adds initial version of an init command

### PR DESCRIPTION
**Pull request by DLMR**
Will currently install a Roc "base" from Github

### CODE REVIEW

**Comment by DLMR** [comment]
We need to write some documentation on how the command works and how a _"base"_ is structured in the future.

File: src/cli/roc-init.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+program
+    .arguments('[base] [version]')
+    .action((argumentOption, version) => {
+        option = argumentOption;
+        versionOption = version;
+    })
+    .parse(process.argv);
+
+// Make sure the directory is empty!
+isEmptyDir();
```
Comment (andreasrs) [inline src/cli/roc-init.js]: For clarity I think we should name this `assertEmptyDir()`

File: src/cli/roc-init.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+/*
+ * Helpers
+ */
+function fetchBase(toFetch, selectVersion) {
+    if (toFetch.indexOf('/') === -1) {
+        const base = bases.find((elem) => elem.identifier === toFetch);
+        if (!base) {
+            console.log('Unvalid name given.');
```
Comment (andreasrs) [inline src/cli/roc-init.js]: Invalid*

File: src/cli/roc-init.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            process.exit(0);
```
Comment (andreasrs) [inline src/cli/roc-init.js]: Imo this should be non-0, if invalid name is to be considered an error

File: src/cli/roc-init.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                console.log(`Could not find the version you wanted, will instead use ${version}`);
+            } else if (!selectedVersion) {
+                console.log(`Using ${version}`);
+            }
+
+            return get(toFetch, version);
+        })
+        .then(() => {
+            if (!validRocProject()) {
+                del.sync(['*', '.*']);
```
Comment (andreasrs) [inline src/cli/roc-init.js]: let's check the safety of this. We will probably want some safety net here 8)

File: src/cli/roc-init.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+function getPrompt() {
+    try {
+        return require(path.join(process.cwd(), 'roc.setup.js')).prompt;
+    } catch (error) {
+        return require('../index').prompt;
+    }
+}
+
+function replaceTempletedValues(answers) {
```
Comment (andreasrs) [inline src/cli/roc-init.js]: I think `Templated` is the correct spelling

File: src/cli/roc-init.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        choices: choices
+    }], answers => {
+        fetchBase(answers.option);
+    });
+}
+
+function isEmptyDir() {
+    if (fs.readdirSync(process.cwd()).length > 0) {
+        console.log('You need to call this command from a empty directory.');
+        process.exit(0);
```
Comment (andreasrs) [inline src/cli/roc-init.js]: Non-0

File: src/cli/helpers/runner.js [hunk 7]
```diff
@@ -2,11 +2,13 @@ import 'source-map-support/register';
...
 import { merge } from 'roc-config';
 
-import { validateRocProject, getBaseRocExtension } from './general';
+import { validRocProject, getBaseRocExtension } from './general';
 import { configManager, setConfig } from './config-manager';
 
 export default function runner(mode) {
-    validateRocProject();
+    if (!validRocProject()) {
+        process.exit(0);
```
Comment (andreasrs) [inline src/cli/helpers/runner.js]: Error here as well?

File: src/cli/roc-init.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+        });
+    });
+}
+
+function configureFiles() {
+    // Rename package.json to .roc for history purposes
+    fs.renameSync(path.join(process.cwd(), 'package.json'), path.join(process.cwd(), '.roc'));
+
+    // Remove setup files
+    del.sync(['**', '!base/**', '!.roc'], { dot: true });
```
Comment (andreasrs) [inline src/cli/roc-init.js]: Same here. I think we should prompt the user with the absolute path before running any sort of recursive deletes

**Comment by andreasrs** [comment]
Exitcodes can be discussed. I think we need to be 100% sure on how file deletes are handled before merging. Prompting user with y/n is probably the best.

Otherwise looks like a good start, loving the github integration :)

**Comment by DLMR** [comment]
Now handles all of the files in a temp directory before they finally are moved to the current working directory. This means that there will not be any issues in relation to file deletions.

**Pull request merged**